daemon/response.go

```diff
@@ -179,8 +181,11 @@ func makeErrorResponder(status int) errorResponder {
 		res := &errorResult{
 			Message: fmt.Sprintf(format, v...),
 		}
-		if status == http.StatusUnauthorized {
+		switch status {
+		case http.StatusUnauthorized:
 			res.Kind = errorKindLoginRequired
+		case http.StatusNotFound:
+			res.Kind = errorKindSnapNotFound
```

Comment:
I'm not understanding this , how do we know it's a not found about a snap instead of some other entity/resouce, the REST api deals with other stuff as well